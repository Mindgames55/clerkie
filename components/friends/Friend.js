import styles from './Friend.module.css'

const STATUS_TO_COPY = {
    close: 'Close Friends',
    'super-close': 'Super Close Friends'
}

export default function Friend({friend}) {
    const {first_name: firstName, last_name: lastName} = friend;
    const fullName = `${firstName} ${lastName}`;

    return (
        <li className={styles.container}>
            <span className={styles.name}>{fullName}</span>
            {friend.status !== 'default' && <span className={styles[friend.status]}>{STATUS_TO_COPY[friend.status]}</span>}
            <span className={styles.metadata}>
                <span className={styles.email}>{friend.email}</span>
                <span>{friend.phone_number}</span>
            </span>
        </li>
    )
}