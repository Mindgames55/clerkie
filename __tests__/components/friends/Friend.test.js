import { render, screen } from '@testing-library/react';
import Friend from '@/components/friends/Friend';
import friendMock from './friend.json'

it('renders Friend correctly', () => {
    const { container } = render(<Friend friend={friendMock}/>);
    expect(container).toMatchSnapshot();
});

it('renders Friend correctly when status is not default', () => {
    render(<Friend friend={{...friendMock, status: 'close'}}/>);
    const span = screen.getByText(/Close Friends/i);
    expect(span).toBeInTheDocument();
});