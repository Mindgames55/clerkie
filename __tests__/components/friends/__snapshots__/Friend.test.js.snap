// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Friend correctly 1`] = `
<div>
  <li
    class="container"
  >
    <span
      class="name"
    >
      Jhon Smith
    </span>
    <span
      class="metadata"
    >
      <span
        class="email"
      >
        email
      </span>
      <span>
        5
      </span>
    </span>
  </li>
</div>
`;
